src/modules/bulk-minter/components/change/ChangeDonationField.tsx

```diff
@@ -0,0 +1,295 @@
+    if (props.creators.length >= MAX_CREATOR_LIMIT) {
+      throw new Error('Max level of creators reached');
+    }
+    const existingCreators = props.creators.map((c) => c.address);
+    const indexOfDuplicate = existingCreators.findIndex(
+      (a) => nonProfit.crypto.solana_address === a
+    );
+    if (indexOfDuplicate !== -1) {
+      throw new Error('All creator hashes must be unique');
+    }
+    const newShareSplit = (100-HOLAPLEX_CREATOR_OBJECT.share) / (props.creators.length + 1);
+    props.updateCreatorState([
+      ...props.creators.map((c) => ({ ...c, share: newShareSplit })),
+      {
+        address: nonProfit.crypto.solana_address,
+        share: newShareSplit,
+        charityProps: {
+          isCharity: true,
+          displayName: nonProfit.name,
+          imageUrl: nonProfit.icon_url,
+        },
+      },
+    ]);
+    setSearchResults([]);
+    props.updateDonationFieldState(false);
+    props.updateShowErrorState(false);
+  }, [setSearchResults, props]);
+
+  const performSearch: (textToSearch: string) => void = useCallback(textToSearch => {
+    setLoading(true);
+    const queryParams = new URLSearchParams();
+    queryParams.append('search_term', textToSearch!);
+    fetch(`https://api.getchange.io/api/v1/nonprofit_basics?${queryParams.toString()}`, {
+      headers: {
+        'Content-Type': 'application/json',
+      },
+    })
+      .then((response) => response.json())
+      .then((response) => {
+        // Some nonprofits do not have crypto addresses; filter these out.
+        return response.nonprofits.filter(
+          (n: ChangeNonprofit) => n.crypto !== undefined
+        ) as ChangeNonprofit[];
+      })
+      .then((nonprofits) => {
+        setSearchResults(nonprofits);
+      })
+      .catch(() => {
+        console.error('error finding nonprofits');
+      })
+      .finally(() => {
+        setIsTyping(false);
+        setLoading(false);
+      });
+  }, [setLoading, setSearchResults, setIsTyping]);
+
+  const handleSearch: (e: React.ChangeEvent<HTMLInputElement>) => void = useCallback(async (e) => {
+    setIsTyping(true);
+    debounce.debounceFcn(() => {
+      if (e.target.value.trim() === '') {
+        setSearchResults([]);
+        setLoading(false);
+        setIsTyping(false);
+      } else {
+        performSearch(e.target.value);
+      }
+    }, DEBOUNCE_INTERVAL_MS);
+  }, [setIsTyping, setSearchResults, setLoading, performSearch]);
+  
+  return (
+    <>
+      {props.showDonationField && (
+        <Row>
+          <Form.Item
+            name="addDonation"
+            style={{ width: '100%', marginBottom: '0px' }}
+            rules={[
+              { required: true, message: 'Please enter a value' },
+              {
+                message: `You can only add ${MAX_CREATOR_LIMIT} creators`,
+                async validator() {
+                  if (props.creators.length === MAX_CREATOR_LIMIT) {
```

Comment:
should this be `if (props.creators.length > MAX_CREATOR_LIMIT) {`? If not, would be good to put a comment that says why this is correct

Reply:
I think you're right, this was just copied from an existing check. Updated both the existing one and this new line.